src/components/dashboard/Dashboard.js

```diff
@@ -568,7 +571,12 @@ const Dashboard = props => {
     [headerLarge, setHeaderLarge],
   )
 
-  const goToProfile = useOnPress(() => screenProps.push('Profile'), [screenProps])
+  // for native we able handle onMomentumScrollEnd, but for web we able to handle only onScroll event,
+  // so we need to imitate onMomentumScrollEnd for web
+  const onScroll = Platform.select({
```

Comment:
hook should be top-level
1. create debounced handleScrollEnd via useMemo 
 ```javascript
const handleScrollEndDebounced = useMemo(() => _debounce(handleScrollEnd, 300), [handleScrollEnd])
```

2. use it inside onScroll:
```javascript
const onScroll = Platform.select({ web: handleScrollEndDebounced, default: noop })
```

Reply:
approved, just fix this one please

Reply:
done